tests/test_box_Box.py

```diff
@@ -112,6 +112,38 @@ def test_wrap_multiple_images(self):
         points = np.array(points)
         npt.assert_allclose(box.wrap(points)[0, 0], -2, rtol=1e-6)
 
+    def test_out_default(self):
+        box = freud.box.Box(2, 2, 2, 1, 0, 0)
+        points = [[10, -5, -5], [0, 0.5, 0]]
+        npt.assert_allclose(box.wrap(points)[0, 0], -2, rtol=1e-6)
+
+    def test_out_provided_with_input_array(self):
+        box = freud.box.Box(2, 2, 2, 1, 0, 0)
+        points = [[10, -5, -5], [0, 0.5, 0]]
+        npt.assert_allclose(box.wrap(points, out=points)[0, 0], -2, rtol=1e-6)
```

Comment:
One way to test the `out` behavior more carefully would be to make the input array read-only. You can do that by setting flags on the array with NumPy. I think that should fail if you let `out` be the default (overwrite in place).

Reply:
I also added a test for this.